test/contrib/test_control_flow.py

```diff
@@ -132,3 +132,65 @@ def iteration(c_prev, c_in):
         result,
         [[1.7, 0.3]],
     )
+
+
+def test_cond():
+    def model():
+        def true_fun(_):
+            x = numpyro.sample("x", dist.Normal(5.0))
+            numpyro.deterministic("z", x - 5.0)
+
+        def false_fun(_):
+            x = numpyro.sample("x", dist.Normal(0.0))
+            numpyro.deterministic("z", x)
+
+        cluster = numpyro.sample("cluster", dist.Normal())
+        cond(cluster > 0, true_fun, false_fun, None)
+
+    def guide():
+        m1 = numpyro.param("m1", 2.5)
+        s1 = numpyro.param("s1", 0.1, constraint=dist.constraints.positive)
+        m2 = numpyro.param("m2", 2.5)
+        s2 = numpyro.param("s2", 0.1, constraint=dist.constraints.positive)
+
+        def true_fun(_):
+            numpyro.sample("x", dist.Normal(m1, s1))
+
+        def false_fun(_):
+            numpyro.sample("x", dist.Normal(m2, s2))
+
+        cluster = numpyro.sample("cluster", dist.Normal())
+        cond(cluster > 0, true_fun, false_fun, None)
+
+    svi = SVI(model, guide, numpyro.optim.Adam(1e-2), Trace_ELBO(num_particles=100))
+    params, losses = svi.run(random.PRNGKey(0), num_steps=2500)
+
+    predictive = Predictive(
+        model,
+        guide=guide,
+        params=params,
+        num_samples=1000,
+        return_sites=["cluster", "x", "z"],
+    )
+    result = predictive(random.PRNGKey(0))
+
+    assert result["cluster"].shape == (1000,)
+    assert result["x"].shape == (1000,)
+    assert result["z"].shape == (1000,)
+
+    mcmc = MCMC(
+        NUTS(model),
+        num_warmup=100,
+        num_samples=2500,
+        num_chains=4,
+        chain_method="sequential",
+    )
+    mcmc.run(random.PRNGKey(0))
+
+    x = mcmc.get_samples()["x"]
+    assert x.shape == (10_000,)
+    assert_allclose(
+        [x[x > 2.5].mean(), x[x > 2.5].std(), x[x < 2.5].mean(), x[x < 2.5].std()],
+        [5.0, 1.0, 0.0, 1.0],
```

Comment:
Could you also check for mixture mean and mixture std: `x.mean() vs 2.5`, `x.std() vs sqrt(2)`?

Reply:
Yep, added, though I made the mixture std `sqrt(29 / 4)`.

```
E(X^2) = 0.5 * (E(X^2 | c = 0) + E(X^2 | c = 1)) = 0.5 * (1 + (1 + 25)) = 27 / 2
```

Then subtract off `2.5^2`. Do you agree? I checked this empirically with NumPy.

Reply:
You are right! I calculated that term wrongly. Thanks!